sugar.go

```diff
@@ -0,0 +1,116 @@
+package vshard_router //nolint:revive
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/tarantool/go-tarantool/v2/pool"
+)
+
+// CallRequest helps you to create a call request object for execution
+// by a Connection.
+type CallRequest struct {
+	ctx      context.Context
+	fnc      string
+	args     interface{}
+	bucketID uint64
+}
+
+// CallResponse is a backwards-compatible structure with go-tarantool for easier replacement.
+type CallResponse struct {
+	rawResp     interface{}
+	getTypedFnc StorageResultTypedFunc
+	err         error
+}
+
+// NewCallRequest returns a new empty CallRequest.
+func (r *Router) NewCallRequest(function string) *CallRequest {
+	req := new(CallRequest)
+	req.fnc = function
+	return req
+}
+
+// Do perform a request synchronously on the connection.
+// It is important that the logic of this method is different from go-tarantool.
+func (r *Router) Do(req *CallRequest, userMode pool.Mode) *CallResponse {
+	ctx := req.ctx
+	bucketID := req.bucketID
+	resp := new(CallResponse)
+
+	if req.fnc == "" {
+		resp.err = fmt.Errorf("func name is empty")
+		return resp
+	}
+
+	if req.args == nil {
+		resp.err = fmt.Errorf("no request args")
+		return resp
+	}
+
+	if req.bucketID == 0 {
+		if r.cfg.BucketGetter == nil {
+			resp.err = fmt.Errorf("bucket id for request is not set")
+			return resp
+		}
+
+		bucketID = r.cfg.BucketGetter(ctx)
+	}
+
+	vshardMode := ReadMode
+
+	if userMode == pool.RW {
+		vshardMode = WriteMode
+	}
+
+	resp.rawResp, resp.getTypedFnc, resp.err = r.RouterCallImpl(ctx,
+		bucketID,
+		CallOpts{
+			Timeout:    r.cfg.RequestTimeout,
```

Comment:
Let's not duplicate configs, we already have timeout config. So we should just use here:
```golang
Timeout: r.cfg.PoolOpts.Timeout
```